libbeat/common/docker/watcher.go

```diff
@@ -125,7 +128,7 @@ func NewWatcherWithClient(client Client, cleanupTimeout time.Duration) (*watcher
 // Container returns the running container with the given ID or nil if unknown
 func (w *watcher) Container(ID string) *Container {
 	w.RLock()
-	container := w.containers[ID]
+	container := w.containers[ID[:truncateLen]]
```

Comment:
I wonder if we should make this a config option or check always for the truncated id. I assume also with 12 digits the likelyhood of overlap is very low.

Reply:
It's an interesting point, *in theory*, chances of collision are round 0.000059%, which makes it very unlikely. That said, someone reported one there:
https://github.com/moby/moby/issues/28260#issuecomment-259820264

Reply:
That's really interesting. I didn't think about it, because I thought it was very-very-very-very uncommon to get a collision. I guess we should consider printing a warning message if we get more than one containers for a specific ID. Thanks for noticing this.

Reply:
My initial thought was that by default we should use the full `id` as there are probably reason the id is that long. The shorter id my assumption is more for user consumption but as it seems it's also around it some other places. 

If a short id is found / provided it should be best effort to match it to a long one with something like `contains`. The problem is that this could become pretty inefficient as it's not a direct match in the array which makes the implementation you did pretty nice.

It would be great if we could find an efficient method to internally store the full id but still match the short id when needed.

Thought: What if we store both?

Reply:
You can have another map `map[string]*Container` that contains short ID as a key.
When we detect that the given ID is in length 12 then we use that map otherwise use the original map.

The disadvantage here is obviously a space-time trade-off.

Reply:
What about keeping current code and make truncate length configurable? That would support the short id use case, but you have to enable it

Reply:
fine by me. @ruflin WDYT?

Reply:
If the short IDs are consistently 12 characters then adding a secondary entry to the map SGTM. But if we need to do a prefix search where the prefixes are of arbitrary length then maybe consider a different data structure.

Reply:
@andrewkroh there are going to be only two kind of IDs:
* short ID (length between 1 and 64 theoretically)
* ID (length of 64)

In other words, the length of a key in the map is `64` or `12`. You won't find a key that its length is `23`.

Reply:
I would also just add a second entry to the map so on the lookup side it still works as before. I would not expect too much overhead as this map will probably not contain more then 1000 entries?

If it becomes an overhead, I would introduce a config option to enable short id's and have it not store in the map if it's off, but still use the same map if it's on. But we can do that if it's causes any issue.